hrm-domain/hrm-service/src/profile/profile-routes-v0.ts

```diff
@@ -24,6 +24,48 @@ import { getCasesByProfileId } from '../case/caseService';
 
 const profilesRouter = SafeRouter();
 
+/**
+ * Returns a filterable list of cases for a helpline
+ *
+ * @param {string} req.accountSid - SID of the helpline
+ * @param {profileController.ProfileListConfiguration['sortDirection']} req.query.sortDirection - Sort direction
+ * @param {profileController.ProfileListConfiguration['sortBy']} req.query.sortBy - Sort by
+ * @param {profileController.ProfileListConfiguration['limit']} req.query.limit - Limit
+ * @param {profileController.ProfileListConfiguration['offset']} req.query.offset - Offset
+ * @param {profileController.SearchParameters['filters']['profileFlagIds']} req.query.profileFlagIds
+ */
+profilesRouter.get('/', publicEndpoint, async (req, res, next) => {
+  try {
+    const { accountSid } = req;
+    const { sortDirection, sortBy, limit, offset, ...rest } = req.query;
+
+    const profileFlagIds = rest.profileFlagIds
+      ? decodeURIComponent(rest.profileFlagIds)
+          .split(',')
+          .map(s => parseInt(s, 10))
+          .filter(v => v && !isNaN(v))
+      : undefined;
+
+    const filters = {
+      profileFlagIds,
+    };
+
+    const result = await profileController.listProfiles(
+      accountSid,
+      { sortDirection, sortBy, limit, offset },
+      { filters },
+    );
+
+    if (isErr(result)) {
+      return next(createError(result.statusCode, result.message));
```

Comment:
Yes, I am aware of this. This is a leftover of the initial works around our `Result` type, where the "status code" was introduced as part of the type itself. I disagree with this idea for the same reasons as you do :stuck_out_tongue: 

My idea around this is to change the status code for a `code`/`name` property, and to start a "dictionary of errors", which will initially look something like `NotFound | NotAllowed | InternalError`. Service layer is aware of the actual reason to raise an error so it will be fair to use this abstraction there (IMO), and then the HTTP layer will have a wrapper around `createError` that takes in an `Err` variant and then converts the error `code` into the proper HTTP status code.
My plan is to do this as part of the "freestyle eng improvements" in one of the next sprints, if time allows.